src/linux_ip_socket/linux_ip_socket.h

```diff
@@ -94,21 +95,14 @@ class linux_ip_socket_private_data final
     void driver_send(const uint8_t* data, const size_t length);
 
   private:
-    enum PacketParseState
-    {
-        STATE_WAIT,
-        STATE_DATA_BYTE,
-        STATE_ESCAPE_BYTE,
-    };
 
     static constexpr int DRIVER_THREAD_PRIORITY = 1;
     static constexpr int DRIVER_THREAD_STACK_SIZE = 65536;
     static constexpr int DRIVER_MAX_CONNECTIONS = 1;
-    static constexpr size_t DRIVER_SEND_BUFFER_SIZE = 8 * 1024;
-    static constexpr size_t DRIVER_RECV_BUFFER_SIZE = 8 * 1024;
-    static constexpr uint8_t START_BYTE = 0x00;
-    static constexpr uint8_t STOP_BYTE = 0xFF;
-    static constexpr uint8_t ESCAPE_BYTE = 0xFE;
+    static constexpr size_t DRIVER_RECV_BUFFER_SIZE = 1 * 1024;
+    static constexpr size_t ENCODED_PACKET_BUFFER_SIZE = 1 * 1024;
```

Comment:
Also, I am not completely sure whether the original code is perfect. The encoded data can be (in the worst case) twice as big as decoded one - decoded does not contain any escaping. Are we sure that these two buffers should be the same size (this is a question, could you please investigate that?)